src/Support/ResolveInfoFieldsAndArguments.php

```diff
@@ -132,15 +139,67 @@ private function foldSelectionSet(SelectionSetNode $selectionSet, int $descend):
         return $fields;
     }
 
-    private function getValue(ArgumentNode $argumentNode)
+    /**
+     * @param ValueNode $value
+     * @return mixed
+     * @throws \Exception
+     */
+    private function getValue(ValueNode $value)
     {
-        $value = $argumentNode->value;
         if ($value instanceof VariableNode) {
             $variableName = $value->name->value;
 
             return $this->info->variableValues[$variableName] ?? null;
         }
+        // Scalar Types
+        if ($value instanceof IntValueNode || $value instanceof FloatValueNode || $value instanceof StringValueNode || $value instanceof BooleanValueNode || $value instanceof EnumValueNode) {
```

Comment:
I think this could be shorten to: 

    if ($value instanceof ValueNode)

Reply:
Valuenode contains the following types

/**
export type ValueNode = VariableNode
| NullValueNode
| IntValueNode
| FloatValueNode
| StringValueNode
| BooleanValueNode
| EnumValueNode
| ListValueNode
| ObjectValueNode
 */

The parameter is ListValueNode or ObjectValueNode is not applicable.

Reply:
@sowork @crissi should we change the order of conditions?

1. check for ListValueNode and ObjectVakueNode
2. Check for ValueNode

This way we would not be concerned anymore about all the details as they effectively work the same.

Another approach for 2.: we effectively treat the "fallthrough" in this method as always being `ValueNode` and return `$argumentNode->value` as it currently is?

Would that make sense?

Reply:
Theoretically feasible, but there is also a nullValueNode

Reply:
not sure

but checking for the ListValueNode and ObjectVakueNode and nullValueNode first

and then fallthough to just get the value might work. So custom scalar types is also considered, not sure the current implementation take this in account.

Reply:
> but there is also a nullValueNode

Right! Forgot to list it, but that would apply for it too.

> So custom scalar types 

Oh, interesting. Didn't think about it and currently don't know what kind of node it exactly is.

We need a test for this too. 😄

Reply:
> custom scalar types

for custom scalar types will still be converted to a  `ValueNode`